cmd/remediation-aggregator/main.go

```diff
@@ -0,0 +1,426 @@
+	if err != nil {
+		return "", err
+	}
+	defer bzReader.Close()
+
+	// FIXME: probably unsafe, see https://haisum.github.io/2017/09/11/golang-ioutil-readall/
+	b, err := ioutil.ReadAll(bzReader)
+	if err != nil {
+		return "", err
+	}
+
+	return string(b), nil
+}
+
+func parseResultRemediations(scheme *runtime.Scheme, scanName, namespace, content string, cm *v1.ConfigMap) ([]*complianceoperatorv1alpha1.ComplianceRemediation, error) {
+	var scanResult string
+	var err error
+
+	_, ok := cm.Annotations[configMapRemediationsProcessed]
+	if ok {
+		fmt.Printf("ConfigMap %s already processed\n", cm.Name)
+		return nil, nil
+	}
+
+	scanResult, ok = cm.Data["results"]
+	if !ok {
+		return nil, fmt.Errorf("no results in configmap %s", cm.Name)
+	}
+
+	_, ok = cm.Annotations[configMapCompressed]
+	if ok {
+		fmt.Printf("Results are compressed\n")
+		scanResult, err = readCompressedData(scanResult)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return utils.ParseRemediationFromContentAndResults(scheme, scanName, namespace, content, scanResult)
+}
+
+// returns true if the lists are the same, false if they differ
+func diffRemediationList(oldList, newList []*complianceoperatorv1alpha1.ComplianceRemediation) bool {
+	if newList == nil {
+		return oldList == nil
+	}
+
+	if len(newList) != len(oldList) {
+		return false
+	}
+
+	sortMcSlice := func(mcSlice []*complianceoperatorv1alpha1.ComplianceRemediation) {
+		sort.SliceStable(mcSlice, func(i, j int) bool { return mcSlice[i].Name < mcSlice[j].Name })
+	}
+
+	sortMcSlice(oldList)
+	sortMcSlice(newList)
+
+	for i := range oldList {
+		ok := diffRemediations(oldList[i], newList[i])
+		if !ok {
+			return false
+		}
+	}
+
+	return true
+}
+
+// returns true if the remediations are the same, false if they differ
+// for now (?) just diffs the MC specs and the remediation type, not sure if we'll ever want to diff more
+func diffRemediations(old, new *complianceoperatorv1alpha1.ComplianceRemediation) bool {
+	if old.Spec.Type != new.Spec.Type {
+		return false
+	}
+
+	// should we be more picky and just compare what can be set with the remediations? e.g. OSImageURL can't
+	// be set with a remediation..
+	return reflect.DeepEqual(old.Spec.MachineConfigContents.Spec, new.Spec.MachineConfigContents.Spec)
+}
+
+func updateResultsConfigMap(clientset *kubernetes.Clientset, cm *v1.ConfigMap) error {
+	cmCopy := cm.DeepCopy()
+
+	if cmCopy.Annotations == nil {
+		cmCopy.Annotations = make(map[string]string)
+	}
+	cmCopy.Annotations[configMapRemediationsProcessed] = ""
+
+	_, err := clientset.CoreV1().ConfigMaps(cmCopy.Namespace).Update(cmCopy)
+	return err
+}
+
+func createRemediations(crClient *complianceCrClient, scan *complianceoperatorv1alpha1.ComplianceScan, remediations []*complianceoperatorv1alpha1.ComplianceRemediation) error {
+	for _, rem := range remediations {
+		fmt.Printf("Creating remediation %s\n", rem.Name)
+		if err := controllerutil.SetControllerReference(scan, rem, crClient.scheme); err != nil {
```

Comment:
should the scan own the remediations? or should it be the suite? (legit question, I don't know).

Reply:
I think a better thing to ask is: Should we support creating Remediations out of Scans or should we support going through the Suite only?